_posts/2020-04-18-schedule-your-functions.md

```diff
@@ -0,0 +1,130 @@
+  - kubernetes
+  - serverless
+  - linux
+  - tutorial
+  - example
+author_staff_member: martin
+dark_background: true
+
+---
+
+In this blog post I will show you how you can run your functions on schedule. We will first set our environment and then we will run the cron-connector with example function which will be executed on a schedule.
+
+One of the things we need very often is our functions to be executed on a schedule. This is where the cron-connector shines. We will showcase how you can use it in this post.
+
+## Motivation
+
+OpenFaaS provides very convenient and flexible way to quickly write and expose your code to be used by external systems. Often times those systems don't have a hook to which you can schedule function execution. Also often times we need to run those functions without external systems, we just want our code to be ran on an interval.
+
+For those purposes [connector-sdk](https://github.com/openfaas-incubator/connector-sdk) was created. It provides the means to quickly adapt different systems to invoke functions. Some of them are [Kafka](https://github.com/openfaas-incubator/kafka-connector), [NATS](https://github.com/openfaas-incubator/nats-connector), [AWS SNS](https://github.com/affix/OpenFaaS-SNS) and [many more](https://docs.openfaas.com/reference/triggers/).
+
+Now we need our functions to to be invoked on an interval. The [cron-connector](https://github.com/openfaas-incubator/cron-connector) exists for those purposes much like Kubernetes provides specific object called `CronJob` which executes code on a schedule or how in Linux we have cron configuration which run scripts on a schedule.
+
+Now lets roll up our sleeves and see how we can execute our OpenFaaS functions on time.
+
+## Prerequisites
+
+Before we start we need a couple of binaries to set up our environment:
+
+* [docker](https://docs.docker.com/get-docker/) - the container runtime used in this post
+* [kind](https://kind.sigs.k8s.io/docs/user/quick-start/) - lightweight local Kubernetes cluster which uses containers as nodes
+* [arkade](https://github.com/alexellis/arkade) - one line installation of applications with which we will install OpenFaaS
+* [faas-cli](https://docs.openfaas.com/cli/install/) - the CLI which communicates with the OpenFaaS gateway
+* [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) - the CLI which communicates with your Kubernetes cluster
+
+## Prepare environment
+
+Before showing you how to run our functions on a schedule, first we need to set up the environment with the tools mentioned above.
+
+### Create cluster
+
+First thing we need is running Kubernetes cluster:
+
+```bash
+kind create cluster
+```
+
+Wait for the installation to finish and verify the namespaces are `Active`:
+
+```bash
+kubectl get ns
+```
+
+### Install OpenFaaS
+
+With `arkade` the installation of OpenFaaS boils down to single line command:
+
+```bash
+arkade install openfaas
+```
+
+Wait for `Active` status on all pods:
```

Comment:
I think the status in the out is `Running` not active.  

I would use `kubectl rollout status -n openfaas deploy/gateway` because it will just block until the gateway is ready, which is exactly what you want and doesn't require as much guessing about the output of the `kubectl get pods`

Reply:
I think your suggestion would be better :+1:

Reply:
Done :heavy_check_mark: